src/ASPHERE/pair_gayberne.cpp

```diff
@@ -531,10 +531,14 @@ void PairGayBerne::read_restart_settings(FILE *fp)
 void PairGayBerne::write_data(FILE *fp)
 {
   for (int i = 1; i <= atom->ntypes; i++)
-    fprintf(fp,"%d %g %g %g %g %g %g %g %g\n",i,
+    if (!atom->chartypesflag) fprintf(fp,"%d %g %g %g %g %g %g %g %g\n",i,
```

Comment:
this is not very maintenance-friendly. why not have a "comment" string, that contains either "" or " # <atom type>", depending on whether "chartypesflag" is set, and then a single `fprintf()`?

Reply:
okay. or how about something llike this?
void PairGayBerne::write_data(FILE *fp)
{
  for (int i = 1; i <= atom->ntypes; i++) {
    fprintf(fp,"%d %g %g %g %g %g %g %g %g",i, ... );
    if (!atom->chartypesflag) fprintf(fp,"\n");
    else fprintf(fp,"# %s\n",atom->char_atomtype[i]);
  }
}

Reply:
okay, I misread your suggestion earlier. I like your idea better, will redo

Reply:
on third thought, do you mind current implementation? avoids allocation of chartype arrays unless needed, and the original fprintf gets a little messy otherwise